### DISCUSSION TITLE
[@types/mapbox__mapbox-sdk] Style.layer is typed as string[] but is a array of objects

### DISCUSSION THREAD

**Discussion by niels-anders** [DefinitelyTyped/DefinitelyTyped#56522] [Issues with a @types package]
Hi mapbox community,

currently the `Layers` property of a `Style` [is typed as `string []`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mapbox__mapbox-sdk/index.d.ts#L1499)

whereas this property is actually an object according to [the mapbox documentation](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/) and [the api response model](https://docs.mapbox.com/api/maps/styles/#example-response-retrieve-a-style). Currently my typescript compiler is complaining when I want to update layer properties of a style that I want to update. 

In the same docs [an example style object](https://docs.mapbox.com/api/maps/styles/#example-style-object) is presented as `"layers": ["{layers}"]` so I wonder if the confusion comes from there. 

Or I am missing something?

Edit: example code below:

```ts
import MapiClient from "@mapbox/mapbox-sdk/lib/classes/mapi-client";
import Styles, { Style } from "@mapbox/mapbox-sdk/services/styles";

const client = new MapiClient({accessToken: 'mytoken'});
const response = Styles(client).getStyle({styleId: 'myStyleId'}).response;
const style = response?.body as Style; // I assume this request returns a `Style` object, could it be something else?
style?.layers.forEach(layer => {
    console.info(layer.id); // Property 'id' does not exist on type 'string'.ts(2339)
});
```

[a comment by an automated account is not shown]

**Comment by jeffbdye** [comment]
Hey @niels-anders! It's likely there's a mistake in the types here in this library or they're out of date in some classes/objects, these are unofficial types maintained separately from the main mapbox-sdk library - feel free to make a PR with the fix and I'll check it out whenever I get a chance!
